nukkit/src/main/java/net/craftingstore/nukkit/listeners/InventoryListener.java

```diff
@@ -0,0 +1,58 @@
+package net.craftingstore.nukkit.listeners;
+
+import cn.nukkit.Player;
+import cn.nukkit.event.EventHandler;
+import cn.nukkit.event.Listener;
+import cn.nukkit.event.inventory.InventoryClickEvent;
+import net.craftingstore.core.models.api.inventory.InventoryItem;
+import net.craftingstore.core.models.api.inventory.types.InventoryItemBackButton;
+import net.craftingstore.core.models.api.inventory.types.InventoryItemCategory;
+import net.craftingstore.core.models.api.inventory.types.InventoryItemCloseButton;
+import net.craftingstore.core.models.api.inventory.types.InventoryItemMessage;
+import net.craftingstore.nukkit.CraftingStoreNukkit;
+import net.craftingstore.nukkit.inventory.CraftingStoreInventoryHolder;
+import net.craftingstore.nukkit.inventory.InventoryBuilder;
+import net.craftingstore.nukkit.inventory.InventoryItemHandler;
+import net.craftingstore.nukkit.inventory.handlers.BackButtonHandler;
+import net.craftingstore.nukkit.inventory.handlers.CategoryItemHandler;
+import net.craftingstore.nukkit.inventory.handlers.CloseButtonHandler;
+import net.craftingstore.nukkit.inventory.handlers.MessageButtonHandler;
+
+import java.util.HashMap;
+
+public class InventoryListener implements Listener {
+    private HashMap<Class<? extends InventoryItem>, InventoryItemHandler> handlers = new HashMap<>();
+    private CraftingStoreNukkit instance;
+    private InventoryBuilder inventoryBuilder;
+
+    public InventoryListener(CraftingStoreNukkit instance) {
+        this.instance = instance;
+        this.inventoryBuilder = new InventoryBuilder(instance);
+        this.handlers.put(InventoryItemBackButton.class, new BackButtonHandler(this.inventoryBuilder));
+        this.handlers.put(InventoryItemCategory.class, new CategoryItemHandler(this.inventoryBuilder));
+        this.handlers.put(InventoryItemCloseButton.class, new CloseButtonHandler());
+        this.handlers.put(InventoryItemMessage.class, new MessageButtonHandler());
+    }
+
+    @EventHandler
+    public void onClick(InventoryClickEvent e) {
```

Comment:
This event never gets executed for me. I am able to move all items out of the inventory.

Reply:
I'm not sure why it's not executing, it's registered.

Also, the FakeInventories library is a plugin, so I've changed it to provided, and it will need to be supplied as a plugin library